File: 01恋上数据结构与算法(Java)/Java版/j07队列/testCircleDeque.java
/*************************************************************************
> File Name       : testCircleDeque.java
> Author          : Harold
> Mail            : 2106562095@qq.com
> Github          : www.github.com/Haroldcc
> Created Time    : 2020年03月07日  10时45分16秒
************************************************************************/

package j07队列;

/**
 * testCircleDeque
 */
public class testCircleDeque {

    public static void main(String[] args) {
        CircleDeque<Integer> qu = new CircleDeque<>();
        for (int i = 0; i < 10; i++) {
            qu.enQueueFront(i + 1);
            qu.enQueueRear(i + 100);
        }
        System.out.println(qu);

        for (int i = 0; i < 3; i++) {
            qu.deQueueFront();
            qu.deQueueRear();
        }

        qu.enQueueFront(11);
        qu.enQueueFront(12);
        System.out.println(qu);
        while (!qu.isEmpty()) {
            System.out.println(qu.deQueueFront());

        }
    }
}

File: 02DataStructuresAndAlgorithms/DataStructures/OldImplementation/linearLIstStructures/Queue/testCircleDeque.cpp
/*************************************************************************
> File Name       : testCircleDeque.cpp
> Author          : Harold
> Mail            : 2106562095@qq.com
> Github          : www.github.com/Haroldcc
> Created Time    : 2020年03月07日  15时49分27秒
************************************************************************/

#include "CircleDeque.h"
#include <iostream>

using namespace std;

int main()
{
    CircleDeque<int> qu;
    //  9 8 7 6 5 4 3 2 1 101 102 103 104 105 106 107 108 109尾
    for (int i = 0; i < 10; i++)
    {
        qu.enQueueFront(i + 1);
        qu.enQueueRear(i + 100);
    }

    cout << qu << endl;

    for (int i = 0; i < 3; i++)
    {
        qu.deQueueFront();
        qu.deQueueRear();
    }

    qu.enQueueFront(11);
    qu.enQueueFront(12);
    cout << qu << endl;
    while (!qu.isEmpty())
    {
        cout << qu.deQueueFront() << endl;
    }

    return 0;
}